weeks/week6/report.md

```diff
@@ -12,85 +12,91 @@
 
 ### Problem 1: Floating point numbers
 ```
-insert your pattern here
+(\d)\.\d+
 ```
 ### Problem 2: Years before 1990
 ```
-insert your pattern here
+.+.19[1-8][0-9].
 ```
 ### Problem 3: Hexadecimal colors
 ```
-insert your pattern here
+#(\w|\d){6}
 ```
 ### Problem 4: Grayscale colors
 ```
-insert your pattern here
+#([a-z0-9]{2}|[A-Z0-9])\1{2,}
 ```
 ### Problem 5: Too long lines
 ```
-insert your pattern here
+.{30,}
 ```
 ### Problem 6: Remove repeating words
 ```
-insert your pattern here
+(\s[\w-']+)\1
+$1
 ```
 ### Problem 7: Match HTML tags
 ```
-insert your pattern here
+<[!/]?\w+\s?(\w+)?>
 ```
 ### Problem 8: Cut numbers two digits after floating point
 ```
-insert your pattern here
+(\d\.\d{2})\d+
+$1
 ```
 ### Problem 9: Digit commas formatting
 ```
-insert your pattern here
+(\d)(?=(\d{3})+\b)
+$1,
 ```
 ### Problem 10: Match lowercase function declarations
 ```
-insert your pattern here
+function(?=\s[a-z])\s\w+.\w?.
 ```
 ### Problem 11: Change date formats
 ```
-insert your pattern here
+(\d{4})-(\d{2})-(\d{2})
+$3.$2.$1
 ```
 ### Problem 12: Validate 24h time format
 ```
-insert your pattern here
+\b([0-1]\d|2[0-3]):[0-5]\d
 ```
 ### Problem 13: Validate AM/PM time format
 ```
-insert your pattern here
+\b(0?\d|1[0-2]):[0-5]\d [AP]M
```

Comment:
I think this is the first time i see the `[AP]M` used, everybody else used `AM|PM` or something like this, this made me chuckle because that's what makes people chuckle when they check a lot of homework and get bored.

Otherwise it's cool, yeah